### PR TITLE
Enable 2FA to cockpit with checkbox

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/6085

### CODE REVIEW

File: root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+
 {
-  if ( -f '/root/.2fa.secret') {
+  if ( -e '/root/.2fa_cockpit.secret') {
       $OUT .= "root\n";
   }
   foreach my $name (glob('/var/lib/nethserver/home/*')) {
       $name =~ s|/var/lib/nethserver/home/||g;
-      next if ( ! -f "/var/lib/nethserver/home/$name/.2fa.secret");
+      next if ( ! -e "/var/lib/nethserver/home/$name/.2fa_cockpit.secret");
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]:
```suggestion
      $secret = "/var/lib/nethserver/home/$name/.2fa_cockpit.secret";
      # check if link is valid
      next if ( lstat $secret and not stat $secret );
```

File: root/etc/e-smith/templates/etc/pam.d/cockpit/50Pam_otp [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # one time password authentication
 auth    [default=1 success=ignore]       pam_listfile.so \
-        onerr=fail  item=user  sense=allow  file=/etc/nethserver/otp.force
-auth    required    pam_oath.so usersfile=/etc/nethserver/otp.secret window=10 digits=6
+        onerr=fail  item=user  sense=allow  file=/etc/nethserver/cockpit.otp.force
+auth    required    pam_oath.so usersfile=/etc/nethserver/cockpit.otp.secret window=10 digits=6
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/pam.d/cockpit/50Pam_otp]: Do not rename this one

File: root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base (hunk 1 above)
Comment (stephdl) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: it does not work with root @gsanchietti  ???? it is supposed to work with another user ?

Comment (gsanchietti) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: Just report the same check also above for user root.

Comment (stephdl) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: could you test the code, it seems it doesn't work like you expected

File: root/etc/e-smith/templates/etc/pam.d/cockpit/50Pam_otp (hunk 2 above)
Comment (stephdl) [reply root/etc/e-smith/templates/etc/pam.d/cockpit/50Pam_otp]: done

File: root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base (hunk 1 above)
Comment (gsanchietti) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: Already tested, but feel free to re-test and change it.
The point is: you must not check only if the link exists, but also the link must not be broken.

Comment (stephdl) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: it seems the behaviour is different between a script and inside the template expansion, however I do not know why

Comment (gsanchietti) [reply root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base]: Your code seems to already correctly check for broken links.

**Pull request merged**